### DISCUSSION TITLE
Getting kubernetes error when trying to deploy using pulumi up

### DISCUSSION THREAD

**Discussion by sbnair** [pulumi/pulumi#9466] [Pulumi Providers]
I get this error and in my aws profile this is the content:

 [default]
region = ca-central-1
 output = json
[profile default]
region = ca-central-1

Diagnostics:
  kubernetes:apps/v1:Deployment (my-block-producer):
    error: 2 errors occurred:
    	* resource default/my-block-producer was successfully created, but the Kubernetes API server reported that it failed to fully initialize or become live: 'my-block-producer' timed out waiting to be Ready
    	* Minimum number of live Pods was not attained
Pls let me know how can i solve this error.

**Comment by hcharley** [comment]
Are you able to confirm in a UI or with kubectl or another tool that the deployment successfully passes its [probes](https://newrelic.com/blog/how-to-relic/kubernetes-health-checks)?

**Comment by sbnair** [comment]
It seems its a pulumi eks cluster, pls see the logs , what could be the issue from this log, can you let me know ?
    Type                                   Name                                        Status       
 +   pulumi:pulumi:Stack                    mina-aws-mina-aws                           creating     
 +   ├─ awsx:x:ec2:Vpc                      mina-vpc                                    created      
 +   pulumi:pulumi:Stack                    mina-aws-mina-aws                           creating..   
 +   │  │  ├─ aws:ec2:Subnet                mina-vpc-private-1                          created      
 +   │  │  ├─ aws:ec2:RouteTable             mina-vpc-private-1                          created      
 +   │  │  └─ aws:ec2:RouteTableAssociation  mina-vpc-private-1                          creating..   
 +   │  ├─ awsx:x:ec2:NatGateway             mina-vpc-1                                       created      
 +   pulumi:pulumi:Stack                     mina-aws-mina-aws                                creating.    
 +   pulumi:pulumi:Stack                     mina-aws-mina-aws                                creating..   
 +   │  │  ├─ aws:ec2:Eip                    mina-vpc-0                                       created      
 +   │  │  └─ aws:ec2:NatGateway             mina-vpc-0                                       creating...  
 +   │  │  └─ aws:ec2:NatGateway             mina-vpc-0                                       creating...  
 +   │  │  └─ aws:ec2:NatGateway             mina-vpc-0                                       creating..   
 +   │  │  └─ aws:ec2:NatGateway             mina-vpc-0                                       created      
 +   pulumi:pulumi:Stack                     mina-aws-mina-aws                                creating..   
 +   pulumi:pulumi:Stack                     mina-aws-mina-aws                                creating     
 +   │  │  ├─ aws:ec2:RouteTable             mina-vpc-public-1                                created      
 +   │  │  ├─ aws:ec2:Subnet                 mina-vpc-public-1                                created      
 +   │  │  ├─ aws:ec2:Route                  mina-vpc-public-1-ig                             created      
 +   │  │  └─ aws:ec2:RouteTableAssociation  mina-vpc-public-1                                     created      
 +   pulumi:pulumi:Stack                     mina-aws-mina-aws                                     creating..   
 +   pulumi:pulumi:Stack                     mina-aws-mina-aws                                     creating.    
 +   │  │  ├─ aws:ec2:Subnet                 mina-vpc-public-0                                     created      
 +   pulumi:pulumi:Stack                     mina-aws-mina-aws                                     creating.    
 +   pulumi:pulumi:Stack                     mina-aws-mina-aws                                     creating...  Warning: 
 +   pulumi:pulumi:Stack                     mina-aws-mina-aws                                     creating...  Warning: 
 +   pulumi:pulumi:Stack                     mina-aws-mina-aws                                     creating...  Warning: 
 +   │  │  ├─ aws:ec2:RouteTable             mina-vpc-private-0                                    created      
 +   pulumi:pulumi:Stack                     mina-aws-mina-aws                                     creating..   walk.go:7
 +   │  │  └─ aws:ec2:Route                  mina-vpc-private-0-nat-0                              created      
 +   pulumi:pulumi:Stack                     mina-aws-mina-aws                                     **creating failed**   
 +   ├─ kubernetes:helm.sh/v2:Chart          block-producer                                        created           
 +   │  ├─ kubernetes:core/v1:Service        my-block-producer                                     **creating faile
 +   │  └─ kubernetes:apps/v1:Deployment     my-block-producer                                     **creating faile
 +   ├─ eks:index:Cluster                    mina-private-chain                                    created           
 +   │  ├─ eks:index:ServiceRole             mina-private-chain-instanceRole                       created         
 +   │  │  ├─ aws:iam:Role                   mina-private-chain-instanceRole-role                  created       
 +   │  │  ├─ aws:iam:RolePolicyAttachment   mina-private-chain-instanceRole-03516f97              created       
 +   │  │  ├─ aws:iam:RolePolicyAttachment   mina-private-chain-instanceRole-e1b295bd              created       
 +   │  │  └─ aws:iam:RolePolicyAttachment   mina-private-chain-instanceRole-3eb088f2              created       
 +   │  ├─ eks:index:ServiceRole             mina-private-chain-eksRole                            created         
 +   │  │  ├─ aws:iam:Role                   mina-private-chain-eksRole-role                       created       
 +   │  │  ├─ aws:iam:RolePolicyAttachment   mina-private-chain-eksRole-4b490823                   created       
 +   │  │  └─ aws:iam:RolePolicyAttachment   mina-private-chain-eksRole-90eb1c99                   created       
 +   │  ├─ pulumi-nodejs:dynamic:Resource    mina-private-chain-cfnStackName                       created         
 +   │  ├─ aws:iam:InstanceProfile           mina-private-chain-instanceProfile                    created         
 +   │  ├─ aws:ec2:SecurityGroup             mina-private-chain-eksClusterSecurityGroup            created         
 +   │  ├─ aws:ec2:SecurityGroupRule         mina-private-chain-eksClusterInternetEgressRule       created         
 +   │  ├─ aws:eks:Cluster                   mina-private-chain-eksCluster                         created         
 +   │  ├─ aws:ec2:SecurityGroup             mina-private-chain-nodeSecurityGroup                  created         
 +   │  ├─ aws:ec2:SecurityGroupRule         mina-private-chain-eksNodeInternetEgressRule          created         
 +   │  ├─ aws:ec2:SecurityGroupRule         mina-private-chain-eksNodeClusterIngressRule          created         
 +   │  ├─ aws:ec2:SecurityGroupRule         mina-private-chain-eksNodeIngressRule                 created         
 +   │  ├─ aws:ec2:SecurityGroupRule         mina-private-chain-eksClusterIngressRule              created         
 +   │  ├─ aws:ec2:SecurityGroupRule         mina-private-chain-eksExtApiServerClusterIngressRule  created         
 +   │  ├─ aws:ec2:LaunchConfiguration       mina-private-chain-nodeLaunchConfiguration            created         
 +   │  ├─ pulumi-nodejs:dynamic:Resource    mina-private-chain-vpc-cni                            created         
 +   │  ├─ pulumi:providers:kubernetes       mina-private-chain-eks-k8s                            created         
 +   │  ├─ kubernetes:core/v1:ConfigMap      mina-private-chain-nodeAccess                         created         
 +   │  ├─ aws:cloudformation:Stack          mina-private-chain-nodes                              created         
 +   │  └─ pulumi:providers:kubernetes       mina-private-chain-provider                           created         
 +   ├─ kubernetes:helm.sh/v2:Chart          sidecar                                               created           
 +   │  └─ kubernetes:apps/v1:Deployment     mina-sidecar                                          created         
 +   └─ kubernetes:core/v1:Secret            mina-secret                                           created           
 
Diagnostics:
  kubernetes:core/v1:Service (my-block-producer):
    error: 2 errors occurred:
    	* resource default/my-block-producer was successfully created, but the Kubernetes API server reported that it failed to fully initialize or become live: 'my-block-producer' timed out waiting to be Ready
    	* Service does not target any Pods. Selected Pods may not be ready, or field '.spec.selector' may not match labels on any Pods
 
  pulumi:pulumi:Stack (mina-aws-mina-aws):
    Warning: apiextensions.k8s.io/v1beta1 CustomResourceDefinition is deprecated in v1.16+, unavailable in v1.22+; use apiextensions.k8s.io/v1 CustomResourceDefinition
 
    walk.go:74: found symbolic link in path: /home/ubuntu/mina/helm/block-producer/Makefile resolves to /home/ubuntu/mina/helm/common/Makefile
 
    error: update failed
 
  kubernetes:apps/v1:Deployment (my-block-producer):
    error: 2 errors occurred:
    	* resource default/my-block-producer was successfully created, but the Kubernetes API server reported that it failed to fully initialize or become live: 'my-block-producer' timed out waiting to be Ready
    	* Minimum number of live Pods was not attained
 
Outputs:
    clusterName: "mina-private-chain-eksCluster-4a7352c"
    kubeconfig : {
        apiVersion     : "v1"
        clusters       : [
            [0]: {
                cluster: {
                    certificate-authority-data: "LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUM1ekNDQWMrZ0F3SUJBZ0lCQURBTkJna3Foa2lHOXcwQkFRc0ZBREFWTVJNd0VRWURWUVFERXdwcmRXSmwKY201bGRHVnpNQjRYRFRJeU1EUXlOakUwTWpreU9Gb1hEVE15TURReU16RTBNamt5T0Zvd0ZURVRNQkVHQTFVRQpBeE1LYTNWaVpYSnVaWFJsY3pDQ0FTSXdEUVlKS29aSWh2Y05BUUVCQlFBRGdnRVBBRENDQVFvQ2dnRUJBS1FkClJ5aU9FOVJLYWVZYzFxYVlGWWhtSWw0UU5PTFFjWHlNa1dadE8yNVQweHFQNzd3RzdWeSt3UTBLU0FDemJDSUIKUjVDNkdtZ2NwcDR4MHoyR0psa1FGKy9HZHo3VFJ4amgvaFBZM0RuamlyUGhrYzdPeGR2KzJlUWErdlk5Vmw1UApqaFdVeWlFUEl2K0Zvdis4S3IwWkJtY3pDVGJDaUpkYVJIZGdtRkRhVzk3c3MwSUFacHdZcWNVZ3dkeDgxcVRkCkl5M2c1YjJFSVhJSis3dVhzWllvVUpRUzNiWWJMQUFUd0s4eklEMHhGaGFBUmJINlVLeFd1RU1wdyswMzlSdnYKamNVaU9XWWxrczQ3ZTFpbFRjeEhSa3lqR1d0c2xWd3U3SjJGaW9KSW5wQ1lyVkVBcXlTWmdIZlN4VmVwU01sSwp1amJVVEhBNzUyU21FOFUwckYwQ0F3RUFBYU5DTUVBd0RnWURWUjBQQVFIL0JBUURBZ0trTUE4R0ExVWRFd0VCCi93UUZNQU1CQWY4d0hRWURWUjBPQkJZRUZLTm92dkNiNkVPeWltWHBnalRsWVo1enR5RnlNQTBHQ1NxR1NJYjMKRFFFQkN3VUFBNElCQVFCVXBvcjV3OExCWmdMVHdRc2R2MVI1aU8rT0ZTRmhQZm5neTg4OFh2eDErU1EyVG5GcQprYi9vQWttN0xzR2dsMC9KSWdXcDIxM3ZhL0FSQVZqdjFBbTdZWDFVR3M0VHNMbVlyZlBYNHhxdmdJYkJlZHZxCmN3R1cwQWRPVGN5VE5RZGVtQVZlak5telBJRGxXNy9lckNkVVhTcGdodkNVZnRsMWlYVUZlaDdITEdMY3JXN1AKRWVEN0EvWkpReitISEJmbjA4c0lrK0pVT24raHF6S0tCeENtR2NxQzZrdk1ORlZPelFKZ3dmeDhpVWRlbVE5SwoyMkZhZzI1ckxIdDVRTzdwRmZiU1BNYWxVMjZKRHNmR0RLOE1XSmZjL2dUNjJQNTlWb25YbWZHYUVFcEdoWU5RCmZJbXhyZWFXalFSc09HTXc4Q24rOUcxb05sQWl0Yzc0TUF3NQotLS0tLUVORCBDRVJUSUZJQ0FURS0tLS0tCg=="
                    server                    : "https://F96947F2E51F87A31E1A0DA52755DA8B.gr7.ca-central-1.eks.amazonaws.com"
                }
                name   : "kubernetes"
            }
        ]
        contexts       : [
            [0]: {
                context: {
                    cluster: "kubernetes"
                    user   : "aws"
                }
                name   : "aws"
            }
        ]
        current-context: "aws"
        kind           : "Config"
        users          : [
            [0]: {
                name: "aws"
                user: {
                    exec: {
                        apiVersion: "client.authentication.k8s.io/v1alpha1"
                        args      : [
                            [0]: "eks"
                            [1]: "get-token"
                            [2]: "--cluster-name"
                            [3]: "mina-private-chain-eksCluster-4a7352c"
                        ]
                        command   : "aws"
                    }
                }
            }
        ]
    }

Resources:
    + 62 created
